apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDbType.ValueConversion.cs

```diff
@@ -198,6 +199,17 @@ internal Value ToProtobufValue(object value, SpannerConversionOptions options)
                     }
                     throw new ArgumentException("Struct parameters must be of type SpannerStruct");
 
+                case TypeCode.Numeric:
+                    if (value is SpannerNumeric spannerNumeric)
+                    {
+                        return new Value {StringValue = spannerNumeric.ToString()};
+                    }
+                    if (value is string str)
+                    {
+                        return new Value {StringValue = SpannerNumeric.Parse(str).ToString()};
+                    }
+                    throw new InvalidCastException("Numeric parameters must be of type SpannerNumeric");
```

Comment:
I think it's worth using ArgumentException here (as in the case above). And if we're supporting string as well, that should be in the exception message.